ledger/src/blockstore.rs

```diff
@@ -1174,30 +1194,32 @@ impl Blockstore {
             .expect("Couldn't fetch from SlotMeta column family")
         {
             // Clear all slot related information
-            self.run_purge(slot, slot, PurgeType::Exact)
-                .expect("Purge database operations failed");
+            let run_purge_result = self.run_purge(slot, slot, PurgeType::Exact);
 
             // Clear this slot as a next slot from parent
-            if let Some(parent_slot) = slot_meta.parent_slot {
-                let mut parent_slot_meta = self
-                    .meta(parent_slot)
-                    .expect("Couldn't fetch from SlotMeta column family")
-                    .expect("Unconfirmed slot should have had parent slot set");
-                // .retain() is a linear scan; however, next_slots should
-                // only contain several elements so this isn't so bad
-                parent_slot_meta
-                    .next_slots
-                    .retain(|&next_slot| next_slot != slot);
-                self.meta_cf
-                    .put(parent_slot, &parent_slot_meta)
-                    .expect("Couldn't insert into SlotMeta column family");
-            }
-            // Reinsert parts of `slot_meta` that are important to retain, like the `next_slots`
-            // field.
+            let parent_purge_result = if let Some(parent_slot) = slot_meta.parent_slot {
+                self.clear_parent_next_slot(parent_slot, slot)
```

Comment:
the parent meta is cleaned up after `run_purge` so the child's `slot_meta` no longer exists. later `slot_meta.clear_unconfirmed_slot()` initialized the child's `slot_meta` as an orphan before inserting it.

`clear_parent_next_slot()` will only error if any of the blockstore functions fail, since we have:
`if let Some(mut parent_slot_meta) = parent_slot_meta`, we won't panic if the parent's meta is not found.

Reply:
essentially i believe we should still panic later if:
- `blockstore.meta(parent_slot).is_err()`
- `blockstore.meta_cf.put(parent_slot, &parent_slot_meta).is_err()`
but not if
- `blockstore.meta(parent_slot).unwrap().is_none()`